parlai/tasks/self_chat/worlds.py

```diff
@@ -5,51 +5,111 @@
 # LICENSE file in the root directory of this source tree.
 
 import random
+from typing import Any, Dict, List
 
-from parlai.core.worlds import DialogPartnerWorld, validate
+from parlai.agents.repeat_label.repeat_label import RepeatLabelAgent
+from parlai.core.agents import Agent
+from parlai.core.worlds import create_task, DialogPartnerWorld, validate
 
 
-class InteractiveWorld(DialogPartnerWorld):
+def load_openers(opt):
+    base_task = opt['task'].split(':')[0]
+    if base_task == 'self_chat':
+        # TODO(2284): Load default openers from s3
+        return
+    print('[ loading conversation openers... ]')
+    # create dummy task so we can get openers from the data
+    task_opt = opt.copy()
+    task_opt['task'] = base_task
+    # default train will loop forever, but evalmode will stop after one epoch
+    if task_opt['datatype'].startswith('train'):
```

Comment:
we might have `train:stream` or `train:ordered` and I'm not sure we want to do this.

I think this snippet is a bit better:
```python
if 'train' in task_opt['datatype'] and 'evalmode' not in task_opt['datatype']:
    task_opt['datatype'] = task_opt['datatype'] + ':evalmode'